src/simulations/optimisation.py

```diff
@@ -600,6 +694,10 @@ def _set_objective(
 
         return p
 
+    def _calculate_reactive_power(self, active_power, power_factor):
```

Comment:
do not create twice the same method. Write it once in `utils` and use everywhere.